modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
 // This module provides an array domain map class that is used to
 // represent rank change slices of another array via a domain.
 //
+// Throughout this file, 'down*' refers to the higher-dimensional
+// array/domain/distribution that's been rank-changed in this view,
+// while 'up*' refers to the lower-dimensional, user-facing view.
+//
+// TODOs for this file:
+//
+// * the dist, dom, and arr classes store a certain amount of
+//   duplicated information -- should these be focused in the higher
+//   classes (dist/dom) and referenced by lower ones (dom/dist)?
+//
+// * I don't think I've done anything to privatize the dist or dom
+//   classes... should I?
```

Comment:
OK, what would help is to put "because they are rank-1" next to the assertion earlier in the file that this only matters for rectangular domains. thanks!